File: leetcode/group2/404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        if not root:
            return 0

        left, right = root.left, root.right
        if left and not left.left and not left.right:
            return left.val + self.sumOfLeftLeaves(right)

        return self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right)

File: leetcode/花花酱题集/404.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function (root) {
  let sum = 0;

  (function (root) {
    if (!root) return;
    const { left, right } = root;
    if (left && !left.left && !left.right) sum += left.val;

    arguments.callee(left);
    arguments.callee(right);
  })(root);
  return sum;
};
